src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -297,12 +301,17 @@ public static LauncherStatus run(final String[] args)
       final Configuration commandLineConf =
           CommandLine.parseToConfiguration(args, commandLineClasses);
       final Injector commandLineInjector = tang.newInjector(commandLineConf);
-      logStartupMessage(commandLineClasses, commandLineInjector);
+      LOGGER.info(genStartupMessage(commandLineClasses, commandLineInjector));
       final String runtimeClassName = commandLineInjector.getNamedInstance(RuntimeClassName.class);
-      final String driverJobSubmissionDirectory =
-          commandLineInjector.isParameterSet(DriverJobSubmissionDirectory.class)
-              ? commandLineInjector.getNamedInstance(DriverJobSubmissionDirectory.class)
-              : null;
+      final String driverJobSubmissionDirectory;
```

Comment:
Hmm.  If I see code full of final variables and then encounter a non-final one, I would be suspicious. I don't want to give future developers that interpretation.